### ISSUE TITLE
New "structure on terraform" interface fails with mexes

### ISSUE THREAD

**Issue by MLanghof** [ZeroK-RTS/Zero-K#562]
Reproduction steps:
With the (always-on) mex-snapping active, manually select a mex from the build menu while holding alt. Click slightly away from a metal spot.

You can now choose the terraform height, but it will be centered on where you clicked, not on the mex spot, thus defeating the purpose.

It will also fail to queue the mex onto the mex spot after the terraform (even if the mex could be built there before and you clicked perfectly centered).

In short, the Mex Placement Handler and the new building-terraform interface don't cooperate.

**Comment by xponen** [comment]
okay, I think I know how to fix this. 
(central_build widget knows how to cooperate with Mex Placement Handler, will copy this)
